### PR TITLE
Move sqlquery receiver logic out to a library for reuse

### PR DESCRIPTION
**Pull request by atoulme**
**Description:** 
Extract sqlquery receiver logic out to `pkg/sqlquery` so it may be reused across components.

**Link to tracking Issue:** 
#13547

**Testing:**
Same unit testing, just moving files.

**Documentation:**
Simple README.md.

### CODE REVIEW

File: pkg/sqlquery/scraper.go [hunk 1]
```diff
@@ -29,25 +29,40 @@ import (
...
+		logger:             logger,
+		dbProviderFunc:     providerFunc,
+		clientProviderFunc: clientProviderFunc,
+	}
+}
 
-func (s scraper) ID() config.ComponentID {
+var _ scraperhelper.Scraper = (*Scraper)(nil)
+
+func (s Scraper) ID() config.ComponentID {
```
Comment (pmcollins) [inline pkg/sqlquery/scraper.go:61]: Can we make these all pointer receivers while we're in here? (this was my mistake -- there's just one at the moment and mixing pointer and non-pointer is bad, apparently)

**Comment by dmitryax** [comment]
Please rebasse

File: pkg/sqlquery/scraper.go (hunk 1 above)
Comment (atoulme) [reply pkg/sqlquery/scraper.go:61]: Done!

File: pkg/sqlquery/config.go [hunk 2]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package sqlqueryreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlqueryreceiver"
+package sqlquery // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/sqlquery"
```
Comment (pmcollins) [inline pkg/sqlquery/config.go:15]: How confident are you that we need to move the receiver config to be part of this library? I feel like the config is a receiver-only thing, and the sql API shouldn't know about it.

Comment (atoulme) [reply pkg/sqlquery/config.go:15]: I think I need to redo this work, based off what I learned from oracledb. This PR is not good enough.

**Comment by atoulme** [comment]
As noted, this PR will get a redo once the #16044 PR is merged. Trying to not rush this.

**Comment by atoulme** [comment]
I'm just not sure this is a good idea anymore, if only because the sqlqueryreceiver has more dependencies to other DB drivers. Closing.

**Pull request closed without merging**